app/namespaces/test_adjustment/test_adjustment_service.py

```diff
@@ -0,0 +1,267 @@
+    # e.g., log odds within 1
+    logit_delta = logit(median_adj_prev) - logit(raw_prev)
+    adjusted_closeto_raw = abs(logit_delta) < 1
+
+    # permit the adjusted result if adjusted and raw are both < 0.5
+    # or if both are > 0.5
+    both_below_maxsmall = (median_adj_prev <= 0.5) and (raw_prev <= 0.5)
+    both_above_minbig = (median_adj_prev >= 0.5) and (raw_prev >= 0.5)
+
+    return (adjusted_closeto_raw or both_below_maxsmall or both_above_minbig)
+
+
+def validate_against_schema(input_payload, schema):
+    try:
+        payload = schema.load(input_payload)
+    except ValidationError as err:
+        return err.messages
+    return payload
+
+
+class ModelParamsSchema(Schema):
+    n_prev_obs = fields.Integer()
+    y_prev_obs = fields.Float()
+    n_se = fields.Integer()
+    y_se = fields.Float()
+    n_sp = fields.Integer()
+    y_sp = fields.Float()
+
+
+class TestAdjHandler:
+    def __init__(self, model_code=testadj_model_code, model_name='testadj_binomial_se_sp',
+                 execution_params={}):
+        self.TESTADJ_MODEL = self.get_stan_model_cache(model_code=model_code, model_name=model_name)
+        self.TESTADJ_MODEL_NAME = model_name
+        # Parse execution params
+        self.n_iter = execution_params.get('n_iter', 4000)
+        self.n_chains = execution_params.get('n_chains', 3)
+        self.return_fit = execution_params.get('return_fit', False)
+        self.n_replicates = execution_params.get('n_replicates', 5)
+        self.trials_lim = execution_params.get('trials_lim', 100)
+        self.modelsets_lim = execution_params.get('modelsets_lim', 3)
+
+    # make sure to gitignore model caches, because the model needs to be compiled separately
+    # on each machine - it is system-specific C++ code
+    def get_stan_model_cache(self, model_code: str, model_name: str = 'anon_model', **kwargs: Dict) -> pystan.StanModel:
+        """Use just as you would `pystan.StanModel`"""
+
+        # Create filepath of cached model
+        code_hash = md5(model_code.encode('ascii')).hexdigest()
+        cache_fn = f'stanmodelcache-{model_name}-{code_hash}.pkl'
+
+        # Try to load cached model
+        try:
+            cached_model = pickle.load(open(cache_fn, 'rb'))
+            print(f"Using cached StanModel at filepath {cache_fn}")
+
+        # Otherwise create the model and cache it
+        except FileNotFoundError:
+            cached_model = pystan.StanModel(model_code=model_code, model_name=model_name, **kwargs)
+            with open(cache_fn, 'wb') as f:
+                pickle.dump(cached_model, f)
+        return cached_model
+
+    def fit_one_pystan_model(self, model_params: Dict) -> Tuple:
+        if model_params['n_sp' < 10] or model_params['n_se' < 10]  \
+                or model_params['y_sp' < 5]  or model_params['y_se' < 5]:
+            adapt_delta = 0.99
+        else:
+            adapt_delta = 0.8
+        fit = self.TESTADJ_MODEL.sampling(data=model_params,
```

Comment:
This is where we can pass in the initial values. The documentation for how to pass in initial values is provided here: search for init. https://mc-stan.org/cmdstanr/reference/model-method-sample.html

A good starting format for the argument to pass in to init is as follows:

```
init_one_chain = {
    'prev': y_prev_obs / n_prev_obs,
    'sens': y_se / n_se,
    'spec': y_sp / n_sp
} # frequentist point estimates for each parameter as initial values 
init = [init_one_chain] * n_chains # pystan requires initial values to be expressed in a list of dicts, one dict for each chain
```

Reply:
I don't get this comment - you want to pass in an additional parameter? What do you mean by pass in initial values? Right now we pass in data, iter, chains, control check_hmc_diagnostics.